### PR TITLE
Unit Tests - scorer

### PR DESCRIPTION
**Pull request by nathannaveen**
- Included unit tests for scorer
- Included test data for scorer
- Included an additional validation for scorer

Signed-off-by: nathannaveen <42319948+nathannaveen@users.noreply.github.com>

### CODE REVIEW

**Comment by nathannaveen** [comment]
@calebbrown, a friendly ping!

File: internal/scorer/algorithm/registry.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ package algorithm
 
 import "fmt"
 
+// TODO: Replace global variable.
+
```
Comment (calebbrown) [inline internal/scorer/algorithm/registry.go:20]: nit: Probably worth removing this comment or extending it to provide some context - I'm not sure what you're thinking here.

File: internal/scorer/scorer.go [hunk 2]
```diff
@@ -27,12 +27,17 @@ import (
...
+
 type Scorer struct {
 	a    algorithm.Algorithm
 	name string
 }
 
 func FromConfig(name string, r io.Reader) (*Scorer, error) {
+	if name == "" {
+		return nil, ErrEmptyName
+	}
```
Comment (calebbrown) [inline internal/scorer/scorer.go:40]: Nice!

**Review by calebbrown — CHANGES_REQUESTED** [review]

File: internal/scorer/algorithm/registry.go (hunk 1 above)
Comment (nathannaveen) [reply internal/scorer/algorithm/registry.go]: I have removed it 😃

File: internal/scorer/scorer.go (hunk 2 above)
Comment (nathannaveen) [reply internal/scorer/scorer.go:40]: Thanks!

**Comment by nathannaveen** [comment]
Closing this PR because of https://github.com/ossf/criticality_score/pull/317.

**Pull request closed without merging**